src/store/initial-state.js

```diff
@@ -64,14 +70,36 @@ export const mergeLocalStorage = (state) => {
  * @param {object} data Data prop passed to App component
  * @param {boolean} applyFixes Whether to override initialState
  */
-export const preparePipelineState = (data, applyFixes) => {
+export const preparePipelineState = (data, applyFixes, expandAllPipelines) => {
   const state = mergeLocalStorage(normalizeData(data));
   if (applyFixes) {
     // Use main pipeline if active pipeline from localStorage isn't recognised
     if (!state.pipeline.ids.includes(state.pipeline.active)) {
       state.pipeline.active = state.pipeline.main;
     }
   }
+
+  // Cater for expandAllPipelines in component props or within flag
+  if (expandAllPipelines) {
```

Comment:
Sorry for butting in but I'd suggest moving this logic inside `normalizeData`. We already have some logic to specifically only mark children of the root node to be visible [here](https://github.com/kedro-org/kedro-viz/blob/bcaa303647febd59ea8c652160d5d52d05e21545/src/store/normalize-data.js#L215-L221), so I'd extend it to mark _all_ nodes to be visible based on a parameter.

Reply:
heya good shout - indeed I was debating whether I should do it there in `normalizeData`, thanks for suggesting and reconfirming this.